File: trunk/java_version/model/src/org/fireflow/model/SubflowTask.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.fireflow.model;

import org.fireflow.model.net.Activity;
import org.fireflow.model.resource.SubWorkflowProcess;

/**
 *
 * @author 非也
 * @version 1.0
 * Created on Mar 15, 2009
 */
public class SubflowTask extends Task{

    protected SubWorkflowProcess subWorkflowProcess = null;

    //subflow Task如何会签？

    public SubflowTask(){
        this.setType(SUBFLOW);
    }

    public SubflowTask(IWFElement parent, String name) {
        super(parent, name);
        this.setType(SUBFLOW);
    }

    /**
     * 返回SUBFLOW类型的任务的子流程信息
     * @return
     * @see org.fireflow.model.reference.SubWorkflowProcess
     */
    public SubWorkflowProcess getSubWorkflowProcess() {
        return subWorkflowProcess;
    }

    public void setSubWorkflowProcess(SubWorkflowProcess subWorkflowProcess) {
        this.subWorkflowProcess = subWorkflowProcess;
    }
}

File: dot-net_version/FireWorkflow.Net/FireWorkflow.Net/Model/SubflowTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using FireWorkflow.Net.Model.Resource;

namespace FireWorkflow.Net.Model
{
    /// <summary>子流程类型的Task。</summary>
    public class SubflowTask : Task
    {
        /// <summary>任务所引用的子流程信息。</summary>
        protected SubWorkflowProcess subWorkflowProcess = null;

        //subflow Task如何会签？

        public SubflowTask()
        {
            this.setType(SUBFLOW);
        }

        public SubflowTask(IWFElement parent, String name)
            : base(parent, name)
        {
            this.setType(SUBFLOW);
        }

        /// <summary>返回SUBFLOW类型的任务的子流程信息。</summary>
        public SubWorkflowProcess getSubWorkflowProcess()
        {
            return subWorkflowProcess;
        }

        public void setSubWorkflowProcess(SubWorkflowProcess subWorkflowProcess)
        {
            this.subWorkflowProcess = subWorkflowProcess;
        }
    }
}
